docs/layer2/layer2_support.md

```diff
@@ -0,0 +1,371 @@
+    find_interface_by_mac() {
+      local mac="$1"
+      for iface in $(ls /sys/class/net/); do
+        iface_mac=$(ethtool -P "$iface" 2>/dev/null | awk '{print $NF}')
+        if [ "$iface_mac" == "$mac" ]; then
+          echo "$iface"
+          return
+        fi
+      done
+      echo "Interface not found for MAC $mac" >&2
+      return 1
+    }
+
+    # Find interface names for eth0 and eth1
+    iface_eth0=$(find_interface_by_mac "$mac_eth0")
+    iface_eth1=$(find_interface_by_mac "$mac_eth1")
+
+    # Replace eth0 and eth1 in the script with the actual interface names
+    sed -i "s/eth0/${iface_eth0}/g" /tmp/final_configuration.sh
+    sed -i "s/eth1/${iface_eth1}/g" /tmp/final_configuration.sh
+
+    # Execute the modified script
+    bash /tmp/final_configuration.sh
+```
+
+The CAPP will use go-templates to substitute the placeholders with appropriate values given by the user.
+
+### Layer 2 Networking Setup by the CAPP Operator
+When provisioning a metal node with Layer 2 networking, the Cluster API Provider (CAPP) Operator will perform the following steps:
+1. **Create a ConfigMap for IP Address Management**: The operator will create a new ConfigMap named <cluster_name-port_name> for each port to manage IP addresses. This ConfigMap is critical for tracking and allocating IP addresses as detailed in the *IP Address Management* section.
+2. **Select an Available IP Address**: CAPP will select an available IP address from the ConfigMap to be assigned to the machine, node, or server being provisioned.
+3. **Generate User-Data Script**: Using Go templates, CAPP will substitute the necessary variables in the user-data script, such as port name, IP address, gateway, and VXLAN. These values are provided by the user through the custom resource definition.
+4. **Submit Device Creation Request**: CAPP will then submit a request to create the device, incorporating the generated user-data script for OS and network configuration.
+5. **Verify Network Configuration**: After the machine or device is successfully provisioned, CAPP will poll the device events API to check whether the network configuration was successful. If not, it will handle the failure or timeout as needed.
+6. **Perform Post-Provisioning Network Operations**: Once the device is provisioned and the network configuration from the user-data script is in place, CAPP will make calls to the /ports API to perform additional operations. These include assigning the VLAN to the port, converting the port to Layer 2 if required, and other necessary configurations.
+
+### Explanation of send_user_state_event Function
+The send_user_state_event function in the script is responsible for sending status updates to the user_state_url fetched from Equinix Metadata API. The Metadata API is a service available on every Equinix Metal server instance that allows the server to access and share various data about itself. Here’s how the function works:
```

Comment:
```suggestion
The send_user_state_event function in the script is responsible for sending status updates to the user_state_url fetched from Equinix Metadata API. The Metadata API is a service available on every Equinix Metal server instance (while in Layer3 or Hybrid mode) that allows the server to access and share various data about itself. Here’s how the function works:
```

Reply:
It would be helpful to link to the docs somewhere in this section: https://deploy.equinix.com/developers/docs/metal/server-metadata/user-state/